src/naive/time/mod.rs

```diff
@@ -580,46 +580,30 @@ impl NaiveTime {
     ///            (from_hms(20, 4, 5), -86_400));
     /// ```
     #[must_use]
-    pub fn overflowing_add_signed(&self, mut rhs: OldDuration) -> (NaiveTime, i64) {
-        let mut secs = self.secs;
-        let mut frac = self.frac;
-
-        // check if `self` is a leap second and adding `rhs` would escape that leap second.
-        // if it's the case, update `self` and `rhs` to involve no leap second;
-        // otherwise the addition immediately finishes.
+    pub const fn overflowing_add_signed(&self, rhs: OldDuration) -> (NaiveTime, i64) {
```

Comment:
There appear to be significant logic changes to `overflowing_add_signed`. Currently, `test_time_overflowing_add` does not have any asserts involving fractional seconds.
https://github.com/chronotope/chrono/blob/v0.4.31/src/naive/time/tests.rs#L117

I strongly recommend adding more asserts to `test_time_overflowing_add` specifically targeting this change in logic.

Reply:
> There appear to be significant logic changes to `overflowing_add_signed`.

True, this method was pretty convoluted because it comes from a time before `rem_euclid`, like we had in more places in chrono.

This code is *extremely* well-tested, better than any other part of chrono. The tests just work on the higher-level methods that wrap `test_time_overflowing_add`.